### PR TITLE
Feature/split operator

### PR DESCRIPTION
**Pull request by dacleyra**
Renaming

- To Redhat Marketplace Deployment Operator or redhat-marketplace-operator in cases to support upgrade
- To Metrics Operator

### CODE REVIEW

File: v2/config/prometheus/monitor.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@ spec:
   endpoints:
     - bearerTokenSecret:
         key: token
-        name: metering-servicemonitor-metrics-reader
+        name: metrics-servicemonitor-metrics-reader
```
Comment (jiangchen2019) [inline v2/config/prometheus/monitor.yaml:13]: metrics-opreator-servicemonitor-metrics-reader?

File: v2/config/prometheus/secret.yaml [hunk 2]
```diff
@@ -1,10 +1,10 @@
-# Secret for metering-servicemonitor-metrics-reader ServiceAccount for ServiceMonitor bearerTokenSecret
+# Secret for metrics-servicemonitor-metrics-reader ServiceAccount for ServiceMonitor bearerTokenSecret
```
Comment (jiangchen2019) [inline v2/config/prometheus/secret.yaml:1]: metrics-operator-servicemonitor-metrics-reader?

File: v2/config/prometheus/secret.yaml [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
 apiVersion: v1
 kind: Secret
 metadata:
   name: servicemonitor-metrics-reader
   labels:
     name: servicemonitor-metrics-reader
   annotations:
-    kubernetes.io/service-account.name: metering-servicemonitor-metrics-reader
+    kubernetes.io/service-account.name: metrics-servicemonitor-metrics-reader
```
Comment (jiangchen2019) [inline v2/config/prometheus/secret.yaml:9]: metrics-operator-servicemonitor-metrics-reader?

File: v2/config/rbac/kustomization.yaml [hunk 4]
```diff
@@ -8,9 +8,9 @@ resources:
   # the auth proxy (https://github.com/brancz/kube-rbac-proxy)
   # which protects your /metrics endpoint.
   - auth_proxy_service.yaml
-  - auth_proxy_role.yaml
-  - auth_proxy_role_binding.yaml
-  - classic/role_binding.yaml
+  # - auth_proxy_role.yaml
+  # - auth_proxy_role_binding.yaml
+  # - classic/role_binding.yaml
```
Comment (jiangchen2019) [inline v2/config/rbac/kustomization.yaml:13]: not sure if these were commented by mistake. They are still in deployer/v2/config/rbac/kustomization.yaml

File: v2/hack/docker/bin/entrypoint [hunk 5]
```diff
@@ -6,7 +6,7 @@ set -e
 
 if ! whoami &>/dev/null; then
   if [ -w /etc/passwd ]; then
-    echo "${USER_NAME:-redhat-marketplace-operator}:x:$(id -u):$(id -g):${USER_NAME:-redhat-marketplace-operator} user:${HOME}:/sbin/nologin" >> /etc/passwd
+    echo "${USER_NAME:-metering-operator}:x:$(id -u):$(id -g):${USER_NAME:-metering-operator} user:${HOME}:/sbin/nologin" >> /etc/passwd
```
Comment (jiangchen2019) [inline v2/hack/docker/bin/entrypoint:9]: -metrics-operator?

**Comment by ab-ibm** [comment]
@dacleyra is this as expected that 'Operator recommended Namespace: redhat-marketplace' instead of openshift-redhat-marketplace?

File: v2/config/rbac/kustomization.yaml (hunk 4 above)
Comment (dacleyra) [reply v2/config/rbac/kustomization.yaml]: classic role binding was moved
auth proxy was not a correct configuration, nor used

**Comment by dacleyra** [comment]
> @dacleyra is this as expected that 'Operator recommended Namespace: redhat-marketplace' instead of openshift-redhat-marketplace?

yes, we have to get out of the openshift- prefixed namespace as per OpenShift v4.12 requirement

**Review by ab-ibm — APPROVED** [review]

File: deployer/v2/config/manifests/bases/CSV_description.md [hunk 6]
```diff
@@ -1,24 +1,24 @@
...
 ### **Upgrade Notice**
 
-The metering and deployment functionality have been seperated into two operators.
-  - The deployment functionality remains as part of this Red Hat Marketplace Operator
-    - The Red Hat Marketplace Operator prerequisites the Red Hat Marketplace Metering Operator
+The Red Hat Marketplace Operator metering and deployment functionality have been seperated into two operators.
+  - The deployment functionality remains as part of this Red Hat Marketplace Deployment Operator
+    - The Red Hat Marketplace Deployment Operator prerequisites the Metrics Operator
     - Admin level functionality and permissions are required for deployment functionality
     - ClusterServiceVersion/redhat-marketplace-operator
```
[the remark this answers is not in the document]
Comment (dacleyra) [reply deployer/v2/config/manifests/bases/CSV_description.md:11]: I think it is important to explain the separation, but if you have suggestions for clearer wording, we can revisit

**Review by jiangchen2019 — APPROVED** [review]

**Pull request merged**